src/index.ts

```diff
@@ -0,0 +1,70 @@
+import express from "express";
+import bodyParser from "body-parser";
+import { createClient } from "redis";
+import "./config";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+import { scheduleJobs } from "./jobs.js";
+import {
+  syncServers,
+  syncHashes,
+  storeMessage,
+  searchByRecipient,
+  searchByHash,
+} from "./util.js";
+
+const app = express();
+app.use(bodyParser.json());
+
+export const redis = createClient({
+  url: process.env.REDIS_URL,
+});
+
+redis.on("error", (err) => console.error("Redis Client Error", err));
+await redis.connect();
+
+scheduleJobs();
+
+// TODO: more error handling and HTTP response codes
+
+app.post("/fetch", async (req, res) => {
+  try {
+    if (req.body.recipient) {
+      console.log("Fetching messages for recipient", req.body.recipient);
+      const messages = await searchByRecipient(req.body.recipient);
+      res.json({ messages });
+    } else if (req.body.hashes) {
+      console.log("Fetching messages by hashes", req.body.hashes);
+      const messages = await Promise.all(req.body.hashes.map(searchByHash));
+      res.json({ messages });
+    } else {
+      res.status(400).send("Bad Request");
```

Comment:
```suggestion
      res.status(400);
```
400 is the error code, unless you are providing more info, having bad request there is redundant

Reply:
It's not redundant that's the how the request gets sent, only setting the status, well only sets the status 😭

Reply:
Actual solution would be to make the text more descriptive, maybe even send more details with json. Not sending anything is worse than sending bad request .